providers/hns/hns_roce_u_verbs.c

```diff
@@ -251,12 +251,33 @@ int hns_roce_u_dealloc_mw(struct ibv_mw *mw)
 	return 0;
 }
 
-static int hns_roce_verify_cq(int *cqe, struct hns_roce_context *context)
+enum {
+	CREATE_CQ_EX_FLAGS = IBV_CREATE_CQ_ATTR_IGNORE_OVERRUN,
+};
+
+enum {
+	CREATE_CQ_SUPPORTED_FLAGS = CREATE_CQ_EX_FLAGS,
+};
+
+enum {
+	CREATE_CQ_SUPPORTED_COMP_MASK = IBV_CQ_INIT_ATTR_MASK_FLAGS,
+};
+
+static int verify_cq_create_attr(struct ibv_cq_init_attr_ex *attr,
+				 struct hns_roce_context *context)
 {
-	if (*cqe < 1 || *cqe > context->max_cqe)
+	if (attr->comp_mask & ~CREATE_CQ_SUPPORTED_COMP_MASK)
+		return -EINVAL;
+
+	if (attr->comp_mask & IBV_CQ_INIT_ATTR_MASK_FLAGS &&
+	    attr->flags & ~CREATE_CQ_SUPPORTED_FLAGS)
+		return -EINVAL;
+
+	if (!attr->cqe || attr->cqe > context->max_cqe)
 		return -EINVAL;
 
```

Comment:
Does this pass the pyverbs test suite? I don't think you can support ibv_create_qp_ex without also supporting wc_flags and ibv_start_poll()/etc.

Reply:
Thanks for you reminder, this commit failed on test_cqex. We'll look into the details of cq_ex.

Reply:
The patch to support cq_ex is dropped from this PR because it may take some time to figure out how to use ibv_create_cq_ex() correctly.